packages/palette/src/elements/Modal/index.tsx

```diff
@@ -0,0 +1,234 @@
+    transform: "translate(-50%, 0%) translateY(50vh)",
+    opacity: 1,
+    onRest: null,
+  })
+  const [visibleContent, setVisibleContent] = useState(null)
+
+  const previousChildren = usePrevious(childrenState)
+  const previousFixedButton = usePrevious(fixedButtonState)
+  const previousHasLogo = usePrevious(hasLogoState)
+  const previousTitle = usePrevious(titleState)
+  const contentAnimation = useSpring(springContentAnimation)
+  const modalAnimation = useSpring(springModalAnimation)
+
+  useEffect(() => {
+    if (!show) {
+      setSpringModalAnimation({
+        transform: "translate(-50%, -50%) translateY(0vh)",
+        opacity: 0,
+        onRest: () => {
+          setSpringModalAnimation({
+            transform: "translate(-50%, 0%) translateY(50vh)",
+            onRest: null,
+            opacity: 0,
+          })
+          onClose()
+        },
+      })
+    } else {
+      setVisibleContent(ModalContent)
+      setSpringModalAnimation({
+        transform: "translate(-50%, -50%) translateY(0vh)",
+        opacity: 1,
+        onRest: null,
+      })
+    }
+  }, [show])
+
+  // Listens for props to change and fades in new content
+  useEffect(() => {
+    if (
+      (previousChildren && children !== previousChildren) ||
+      (previousFixedButton &&
+        previousFixedButton.props &&
+        FixedButton &&
+        FixedButton.props.children !== previousFixedButton.props.children) ||
+      (previousTitle && title !== previousTitle) ||
+      (previousHasLogo && hasLogo !== previousHasLogo)
+    ) {
+      setSpringContentAnimation({
+        opacity: 0,
+        onRest: () => updateVisibleContent(),
+      })
+    }
+  }, [hasLogo, title, FixedButton, children])
+
+  const ModalContent = () => {
+    return (
+      <AnimatedView style={contentAnimation}>
+        <ModalScrollContent px={4} pt={2} FixedButton={FixedButton}>
+          {hasLogo && (
+            <>
+              <Flex justifyContent="center">
+                <Logo />
+              </Flex>
+              <Spacer mb={space(1)} />
+            </>
+          )}
+          {title && (
+            <>
+              <Flex justifyContent="center">
+                <Serif size="5t" textAlign="center" color="black100">
+                  {title}
+                </Serif>
+              </Flex>
+              <Spacer mb={space(1)} />
+            </>
+          )}
+          {children}
+        </ModalScrollContent>
+        {FixedButton && (
+          <FixedButtonWrapper px={4} pt={2} pb={3}>
+            {FixedButton}
+          </FixedButtonWrapper>
+        )}
+      </AnimatedView>
+    )
+  }
+
+  const updateVisibleContent = () => {
+    setChildrenState(children)
+    setTitleState(title)
+    setFixedButtonState(FixedButton)
+    setHasLogoState(hasLogo)
+    // Replaces content
+    setVisibleContent(ModalContent)
+    // Fades in new content
+    setSpringContentAnimation({
+      opacity: 1,
+      onRest: null,
+    })
+  }
```

Comment:
This function can be moved up, closer to where it's called. Then the renderable content (`ModalContent`, `return (...)`) can be grouped nicely side-by-side.